### ISSUE TITLE
Procedures Stability Test Failure

### ISSUE THREAD

**Issue by maxwelllls** [pyri-project/pyri-robotics#6]
When conducting stability tests by cyclically invoking Procedures through the "Main" Panel, the process stops after approximately 1000 cycles, with the following error messages appearing in the terminal.

![image](https://github.com/pyri-project/pyri-robotics/assets/51860539/28288a98-8a42-40e0-ad63-612d87f64c3c)
![image](https://github.com/pyri-project/pyri-robotics/assets/51860539/d7dff997-56db-4b40-825a-f242ce5f1632)

Clicking the "run" button in the "new" Procedures separately also fails to execute, accompanied by the following message.
![image](https://github.com/pyri-project/pyri-robotics/assets/51860539/2cf389bb-16fb-4460-89b6-e5bfaa85944d)

**Comment by maxwelllls** [comment]
After numerous tests, it seems that the issue does not only occur with a specific component. Similar phenomena have been observed with the robotics_motion, sandbox, and variable_storage devices. This suggests that the problem may be related to the stability of Robot Raconteur.

**Comment by maxwelllls** [comment]
Hi @johnwason ,sorry to bother you again. Even after fixing the deadlock issue with the RR components, stability testing using the "main" panel still fails. The robotics_motion, sandbox, and variable_storage devices may unexpectedly stop working. 

The method of reproduction remains the same: rapid cycling of tasks, resulting in a crash within a few minutes.

**Comment by johnwason** [comment]
> sorry to bother you again.

Your feedback is much appreciated!

> Even after fixing the deadlock issue with the RR components, stability testing using the "main" panel still fails. The robotics_motion, sandbox, and variable_storage devices may unexpectedly stop working.

Ok, I will keep running duration tests to see if I can reproduce.

**Comment by johnwason** [comment]
Hi @maxwelllls I am aware of this issue, and am fixing some features upstream in the core libraries that should help significantly with this type of stability problem. Thank you for your patience.

**Comment by maxwelllls** [comment]
Appreciate your work on this issue. Eagerly awaiting the fix!

**Comment by johnwason** [comment]
I believe I have fixed this error, and will push out the updates over the next few days along with some overall improvements to the software.

During endurance testing I discovered another issue with variable_storage. I am going to continue endurance testing to figure out why it is crashing, but that will take some time since the error doesn't appear immediately.

**Comment by johnwason** [comment]
I believe I have fixed the errors you encountered, and the problem with variable_storage.

Please reinstall PyRI using the instructions at https://github.com/pyri-project/pyri-core/blob/master/README.md

Use a new venv if possible.

You may need to delete the contents of `$HOME/.local/share/pyri-project` to completely remove the old version.

**Comment by johnwason** [comment]
Has this been resolved?

**Comment by maxwelllls** [comment]
> Has this been resolved?

Apologies for not providing timely feedback. After updating the program, I conducted a retest, running it for about 10 hours and several tens of thousands of cycles, and no errors occurred with pyri. 

However, the realman_robot_driver that I developed myself currently has some stability issues and stops responding after about ten hours. At present, there is reason to believe that the stability issue with pyri has been resolved.

PS: @Lafatalite is our team’s new member, and he will be mainly responsible for the development of the teach pendant in the future.
